packages/engine/src/scene/components/ModelComponent.tsx

```diff
@@ -255,7 +256,9 @@ export const useMeshOrModel = (entity: Entity) => {
   const isModel = !!useOptionalComponent(entity, ModelComponent)
   const isChildOfModel = !!useAncestorWithComponent(entity, ModelComponent)
   const hasMesh = !!useOptionalComponent(entity, MeshComponent)
-  return isModel && !isChildOfModel && hasMesh
+  const isPrimitiveGeom = !!useOptionalComponent(entity, PrimitiveGeometryComponent)
```

Comment:
hmm, i think the original logic here is almost correct, and was changed recently here
https://github.com/EtherealEngine/etherealengine/pull/10262

I think the correct logic here should return when the entity is either:
- mesh that is not a part of a model
- a model

```ts

export const useMeshOrModel = (entity: Entity) => {
  const isModel = !!useOptionalComponent(entity, ModelComponent)
  const isChildOfModel = !!useAncestorWithComponent(entity, ModelComponent)
  const hasMesh = !!useOptionalComponent(entity, MeshComponent)
  return isModel || (!isChildOfModel && hasMesh)
}
```

Reply:
primitive geometry is also a type of model, do we not need to check for that?

Reply:
That should be covered by the condition of `!model && mesh`, which is the point of this complex hook

Reply:
@SYBIOTE can we try make this work without checking for the primitive geom component here

Reply:
We certainly can But do we not want drop shadows for the primitive geom component?

Reply:
> That should be covered by the condition of `!model && mesh`, which is the point of this complex hook

I don't think I remember it covering primitive geom, ill revert it and test it